test/imp/sum-save-proofs-spec.k.save-proofs.kore.golden

```diff
@@ -145,64 +145,135 @@ module haskell-backend-saved-claims-43943e50-f723-47cd-99fd-07104d664c6d
+                                        \dv{SortId{}}(/* Fl Fn D Sfa Cl */ "n")
+                                    ),
+                                    /* Fl Fn D Sfa Cli */
+                                    /* Inj: */ inj{SortInt{}, SortKItem{}}(
+                                        /* Fl Fn D Sfa Cl */
+                                        /* builtin: */ \dv{SortInt{}}("0")
+                                    )
                                 ),
-                                /* Fl Fn D Sfa Cli */
-                                /* Inj: */ inj{SortInt{}, SortKItem{}}(
-                                    /* Fl Fn D Sfa Cl */
-                                    /* builtin: */ \dv{SortInt{}}("0")
+                                /* Fl Fn D Spa */
+                                Lbl'UndsPipe'-'-GT-Unds'{}(
+                                    /* Fl Fn D Sfa Cli */
+                                    /* Inj: */ inj{SortId{}, SortKItem{}}(
+                                        /* Fl Fn D Sfa Cl */
+                                        \dv{SortId{}}(/* Fl Fn D Sfa Cl */
+                                        "sum")
+                                    ),
+                                    /* Fl Fn D Spa */
+                                    /* Inj: */ inj{SortInt{}, SortKItem{}}(
+                                        /* Fl Fn D Spa */
+                                        Lbl'UndsPlus'Int'Unds'{}(
+                                            /* Fl Fn D Sfa */ VarS:SortInt{},
+                                            /* Fl Fn D Spa */
+                                            Lbl'UndsStar'Int'Unds'{}(
+                                                /* Fl Fn D Spa */
+                                                Lbl'UndsPlus'Int'Unds'{}(
+                                                    /* Fl Fn D Sfa */
+                                                    VarN:SortInt{},
+                                                    /* Fl Fn D Sfa Cl */
+                                                    /* builtin: */
+                                                    \dv{SortInt{}}("1")
+                                                ),
+                                                /* Fl Fn D Sfa */ VarN:SortInt{}
+                                            )
+                                        )
+                                    )
                                 )
-                            ),
+                            )
+                        )
+                    ),
+                    /* Fl Fn D Spa */
+                    Lbl'-LT-'generatedCounter'-GT-'{}(
+                        /* Fl Fn D Sfa */ Var'Unds'DotVar3:SortInt{}
+                    )
+                ),
+                /* Spa */
+                \ceil{SortGeneratedTopCell{}, SortGeneratedTopCell{}}(
+                    /* Fn Spa */
+                    Lbl'-LT-'generatedTop'-GT-'{}(
```

Comment:
I know this hasn't been simplified because `Lbl'-LT-'generatedTop'-GT-'{}` is a functional constructor.